src/solver/impls/arkode/arkode_mri.cxx

```diff
@@ -0,0 +1,990 @@
+    throw BoutException("ARKodeSetUserData failed\n");
+  }
+
+  if(inner_treatment != MRI_Treatment::Explicit)
+    if (ARKodeSetLinear(inner_arkode_mem, inner_set_linear) != ARK_SUCCESS) {
+      throw BoutException("ARKodeSetLinear failed\n");
+    }
+
+  if (fixed_step) {
+    // If not given, default to adaptive timestepping
+    const BoutReal inner_fixed_timestep = (*options)["inner_timestep"];
+    if (ARKodeSetFixedStep(inner_arkode_mem, inner_fixed_timestep) != ARK_SUCCESS) {
+      throw BoutException("ARKodeSetFixedStep failed\n");
+    }
+  }
+
+  if (ARKodeSetOrder(inner_arkode_mem, order) != ARK_SUCCESS) {
+    throw BoutException("ARKodeSetOrder failed\n");
+  }
+
+  if (ARKStepCreateMRIStepInnerStepper(inner_arkode_mem, &inner_stepper) != ARK_SUCCESS) {
+    throw BoutException("ARKStepCreateMRIStepInnerStepper failed\n");
+  }
+
+  // Initialize the slow integrator. Specify the explicit slow right-hand side
+  // function in y'=fe(t,y)+fi(t,y)+ff(t,y), the inital time T0, the
+  // initial dependent variable vector y, and the fast integrator.
+
+  switch (treatment) {
+  case MRI_Treatment::ImEx:
+    arkode_mem = callWithSUNContext(MRIStepCreate, suncontext, arkode_rhs_s_explicit, arkode_rhs_s_implicit,
+                                    simtime, uvec, inner_stepper);
+    output_info.write("\tUsing ARKODE-MRI ImEx solver \n");
+    break;
+  case MRI_Treatment::Explicit:
+    arkode_mem = callWithSUNContext(MRIStepCreate, suncontext, arkode_s_rhs, nullptr,
+                                    simtime, uvec, inner_stepper);
+    output_info.write("\tUsing ARKODE-MRI Explicit solver \n");
+    break;
+  case MRI_Treatment::Implicit:
+    arkode_mem = callWithSUNContext(MRIStepCreate, suncontext, nullptr, arkode_s_rhs,
+                                    simtime, uvec, inner_stepper);
+    output_info.write("\tUsing ARKODE-MRI Implicit solver \n");
+    break;
+  default:
+    throw BoutException("Invalid treatment: {}\n", toString(treatment));
+  }
+  if (arkode_mem == nullptr) {
+    throw BoutException("MRIStepCreate failed\n");
+  }
+
+  // For callbacks, need pointer to solver object
+  if (ARKodeSetUserData(arkode_mem, this) != ARK_SUCCESS) {
+    throw BoutException("ARKodeSetUserData failed\n");
+  }
+
+  if(treatment != MRI_Treatment::Explicit)
+    if (ARKodeSetLinear(arkode_mem, set_linear) != ARK_SUCCESS) {
+      throw BoutException("ARKodeSetLinear failed\n");
+    }
+
+  if (fixed_step) {
+    // If not given, default to adaptive timestepping
+    const BoutReal fixed_timestep = (*options)["timestep"];
+    if (ARKodeSetFixedStep(arkode_mem, fixed_timestep) != ARK_SUCCESS) {
+      throw BoutException("ARKodeSetFixedStep failed\n");
+    }
+  }
+
+  if (ARKodeSetOrder(arkode_mem, order) != ARK_SUCCESS) {
+    throw BoutException("ARKodeSetOrder failed\n");
+  }
+
+  if (ARKodeSetFixedStep(arkode_mem, 0.001) != ARK_SUCCESS) {
+    throw BoutException("ARKodeSetFixedStep failed\n");
+  }
+
```

Comment:
This should be removed now that SUNDIALS supports MRI adaptivity (assuming that this MRI interface requires SUNDIALS v7.2.0 or higher, as recommended earlier).
```suggestion
```